File: Project/TeamA_Enterprise-Learning-Platform-main (1)/TeamA_Enterprise-Learning-Platform-main/skillsphere-frontend/src/services/CourseContentService.js
import apiClient from '../api/apiClient';

export const CourseContentService = {
    // ------------------------------
    // Course Module Methods
    // ------------------------------
    getModulesForCourse: async (courseId) => {
        const response = await apiClient.get(`/api/courses/${courseId}/modules`);
        return response.data;
    },

    getModuleById: async (courseId, moduleId) => {
        const response = await apiClient.get(`/api/courses/${courseId}/modules/${moduleId}`);
        return response.data;
    },

    createModule: async (courseId, moduleData) => {
        const response = await apiClient.post(`/api/mentor/courses/${courseId}/modules`, moduleData);
        return response.data;
    },

    updateModule: async (courseId, moduleId, moduleData) => {
        const response = await apiClient.put(`/api/mentor/courses/${courseId}/modules/${moduleId}`, moduleData);
        return response.data;
    },

    deleteModule: async (courseId, moduleId) => {
        await apiClient.delete(`/api/mentor/courses/${courseId}/modules/${moduleId}`);
    },

    reorderModules: async (courseId, moduleIds) => {
        const response = await apiClient.put(`/api/mentor/courses/${courseId}/modules/reorder`, moduleIds);
        return response.data;
    },

    // ------------------------------
    // Lesson Methods
    // ------------------------------
    getLessonsForModule: async (courseId, moduleId) => {
        const response = await apiClient.get(`/api/courses/${courseId}/modules/${moduleId}/lessons`);
        return response.data;
    },

    getLessonById: async (courseId, moduleId, lessonId) => {
        const response = await apiClient.get(`/api/courses/${courseId}/modules/${moduleId}/lessons/${lessonId}`);
        return response.data;
    },

    createLesson: async (courseId, moduleId, lessonData) => {
        const response = await apiClient.post(`/api/mentor/courses/${courseId}/modules/${moduleId}/lessons`, lessonData);
        return response.data;
    },

    updateLesson: async (courseId, moduleId, lessonId, lessonData) => {
        const response = await apiClient.put(`/api/mentor/courses/${courseId}/modules/${moduleId}/lessons/${lessonId}`, lessonData);
        return response.data;
    },

    deleteLesson: async (courseId, moduleId, lessonId) => {
        await apiClient.delete(`/api/mentor/courses/${courseId}/modules/${moduleId}/lessons/${lessonId}`);
    },

    reorderLessons: async (courseId, moduleId, lessonIds) => {
        const response = await apiClient.put(`/api/mentor/courses/${courseId}/modules/${moduleId}/lessons/reorder`, lessonIds);
        return response.data;
    },

    // ------------------------------
    // Resource Methods
    // ------------------------------
    getResourcesForLesson: async (courseId, moduleId, lessonId) => {
        const response = await apiClient.get(`/api/courses/${courseId}/modules/${moduleId}/lessons/${lessonId}/resources`);
        return response.data;
    },

    getResourceById: async (courseId, moduleId, lessonId, resourceId) => {
        const response = await apiClient.get(`/api/courses/${courseId}/modules/${moduleId}/lessons/${lessonId}/resources/${resourceId}`);
        return response.data;
    },

    createResource: async (courseId, moduleId, lessonId, resourceData) => {
        const response = await apiClient.post(`/api/mentor/courses/${courseId}/modules/${moduleId}/lessons/${lessonId}/resources`, resourceData);
        return response.data;
    },

    updateResource: async (courseId, moduleId, lessonId, resourceId, resourceData) => {
        const response = await apiClient.put(`/api/mentor/courses/${courseId}/modules/${moduleId}/lessons/${lessonId}/resources/${resourceId}`, resourceData);
        return response.data;
    },

    deleteResource: async (courseId, moduleId, lessonId, resourceId) => {
        await apiClient.delete(`/api/mentor/courses/${courseId}/modules/${moduleId}/lessons/${lessonId}/resources/${resourceId}`);
    },

    reorderResources: async (courseId, moduleId, lessonId, resourceIds) => {
        const response = await apiClient.put(`/api/mentor/courses/${courseId}/modules/${moduleId}/lessons/${lessonId}/resources/reorder`, resourceIds);
        return response.data;
    }
};

export default CourseContentService;

File: Project/TeamA_Enterprise-Learning-Platform-main (1)/TeamA_Enterprise-Learning-Platform-main/skillsphere-backend/src/main/java/com/skillsphere/service/CourseContentService.java
package com.skillsphere.service;

import com.skillsphere.dto.CourseModuleRequest;
import com.skillsphere.dto.CourseModuleResponse;
import com.skillsphere.dto.LessonRequest;
import com.skillsphere.dto.LessonResponse;
import com.skillsphere.dto.ResourceRequest;
import com.skillsphere.dto.ResourceResponse;
import com.skillsphere.entity.User;

import java.util.List;

public interface CourseContentService {
    // CourseModule methods
    CourseModuleResponse createModule(Long courseId, CourseModuleRequest request, User mentor);
    CourseModuleResponse updateModule(Long courseId, Long moduleId, CourseModuleRequest request, User mentor);
    void deleteModule(Long courseId, Long moduleId, User mentor);
    List<CourseModuleResponse> reorderModules(Long courseId, List<Long> moduleIds, User mentor);
    List<CourseModuleResponse> getModulesForCourse(Long courseId, User user);
    CourseModuleResponse getModuleById(Long courseId, Long moduleId, User user);

    // Lesson methods
    LessonResponse createLesson(Long courseId, Long moduleId, LessonRequest request, User mentor);
    LessonResponse updateLesson(Long courseId, Long moduleId, Long lessonId, LessonRequest request, User mentor);
    void deleteLesson(Long courseId, Long moduleId, Long lessonId, User mentor);
    List<LessonResponse> reorderLessons(Long courseId, Long moduleId, List<Long> lessonIds, User mentor);
    List<LessonResponse> getLessonsForModule(Long courseId, Long moduleId, User user);
    LessonResponse getLessonById(Long courseId, Long moduleId, Long lessonId, User user);

    // Resource methods
    ResourceResponse createResource(Long courseId, Long moduleId, Long lessonId, ResourceRequest request, User mentor);
    ResourceResponse updateResource(Long courseId, Long moduleId, Long lessonId, Long resourceId, ResourceRequest request, User mentor);
    void deleteResource(Long courseId, Long moduleId, Long lessonId, Long resourceId, User mentor);
    List<ResourceResponse> reorderResources(Long courseId, Long moduleId, Long lessonId, List<Long> resourceIds, User mentor);
    List<ResourceResponse> getResourcesForLesson(Long courseId, Long moduleId, Long lessonId, User user);
    ResourceResponse getResourceById(Long courseId, Long moduleId, Long lessonId, Long resourceId, User user);
}
